music21/romanText/writeRoman.py

```diff
@@ -0,0 +1,316 @@
+        (corresponding directly to the structure for the time signatures).
+        '''
+
+        currentMeasure = -100  # i.e. fake
+        currentString = None
+        currentKey = ''
+
+        for rn in self.romanNumerals:
+            keyString = str(rn.key).split(' ')[0].replace('-', 'b')
+            if keyString != currentKey:
+                chordString = f'{keyString}: {rn.figure}'
+                currentKey = keyString
+            else:
+                chordString = str(rn.figure)
+
+            if rn.measureNumber == currentMeasure:  # continue an existing string
+                currentString = rnString(measure=rn.measureNumber,
+                                         beat=rn.beat,
+                                         chordString=chordString,
+                                         inString=currentString)
+            else:
+                self.analysisDict[currentMeasure] = currentString  # Save previous string ...
+                currentString = rnString(measure=rn.measureNumber,  # ... and start a new one.
+                                         beat=rn.beat,
+                                         chordString=chordString)
+            currentMeasure = rn.measureNumber
+
+        # Special case of last entry.
+        self.analysisDict[currentMeasure] = currentString
+
+    def prepList(self):
+        '''
+        Prepares a sequential list of text lines, integrating the timeSignatures and romanNumerals
+        from their respective dicts.
+        I.e. this prepares all lines other than the metadata preamble (handled separately)
+        '''
+
+        tsMeasures = self.timeSigMeasureDict.keys()
+
+        if not self.analysisDict:
+            self.makeMeasureStrings()
+
+        for x in range(self.firstMeasureNumber, self.lastMeasureNumber + 1):
+
+            if x in tsMeasures:  # First, before corresponding measure string
+                ts = self.timeSigMeasureDict[x]
+                self.combinedList.append(f'\nTime Signature: {ts}')
+
+            if x in self.analysisDict.keys():
+                self.combinedList.append(self.analysisDict[x])
+
+# ------------------------------------------------------------------------------
+
+    # To write
+
+    def writeRomanText(self,
+                       outPath: str = '.',
+                       fileName: str = ''):
+        '''
+        Writes the combined information to a 'Roman text' file (using the extension .txt).
+        '''
+
+        if not self.combinedList:
+            self.prepList()
+
+        if not fileName:  # Never an empty string: placeholders set by prepPreamble as needed.
+            fileName = f'{self.composer}_-_{self.title}'
+            fileName = fileName.replace('.', '')
+            fileName = fileName.replace(':', '')
+            fileName = fileName.replace(' ', '_')
+
+        text_file = open(os.path.join(outPath, f'{fileName}.txt'), "w")
+        [text_file.write(entry + "\n") for entry in self.preamble]
```

Comment:
Hi Mark. Pylint will be happier if you don't wrap this in a list comprehension. It's seeing you created a list with `[]` and didn't assign it to a variable.

It's not really more verbose to just break this out into:
```
for entry in self.preamble:
    text_file.write(entry + '\n')
```

Reply:
Thanks! I'll do that.

Reply:
Exactly.  And Myke would be happier too.  It creates a list of Nones that is assigned to nothing so it's memory inefficient.

Reply:
How many `None`s could a NoneCheck check if a `None`...